### PR TITLE
fix: Make tests for sources to be picked up correctly

### PR DESCRIPTION
**Pull request by chamini2**
<details>
closes FEA-192
</details>

### CODE REVIEW

File: src/faldbt/project.py [hunk 1]
```diff
@@ -65,25 +65,43 @@ def set_status(self, status: str):
...
+                if refs:
+                    self.model = refs[0]
+
+                print(metadata.kwargs["model"])
+                sources = re.findall(
+                    r"source\('([^']+)', *'([^']+)'\)", metadata.kwargs["model"]
+                )
+                print(sources)
+                if sources:
+                    self.source = sources[0]
```
Comment (mederka) [inline src/faldbt/project.py:96]: What does this tuple look like?

File: src/faldbt/project.py [hunk 2]
```diff
@@ -65,25 +65,43 @@ def set_status(self, status: str):
...
-                self.column = node.test_metadata.kwargs.get("column_name", None)
+                self.column = metadata.kwargs.get("column_name", None)
+
+                # TODO: Handle package models?
+                refs = re.findall(r"ref\('([^']+)'\)", metadata.kwargs["model"])
+
+                if refs:
+                    self.model = refs[0]
+
+                print(metadata.kwargs["model"])
```
Comment (mederka) [inline src/faldbt/project.py:90]: Probably want to remove print statements

File: src/faldbt/project.py [hunk 3]
```diff
@@ -65,25 +65,43 @@ def set_status(self, status: str):
...
+                sources = re.findall(
```
Comment (mederka) [inline src/faldbt/project.py:91]: So subquery can have `source` as well? Sorry didn't see this.

**Review by mederka — COMMENTED** [review]
Thanks you for looking into this, @chamini2! The PR looks good, just small questions.

File: src/faldbt/project.py (hunk 1 above)
Comment (chamini2) [reply src/faldbt/project.py:96]: `sources` looks like
```
[('results', 'some_source')]
```

**Review by mederka — APPROVED** [review]

File: src/faldbt/project.py [hunk 4]
```diff
@@ -250,15 +266,18 @@ def _map_nodes(
...
                 node=node,
                 _status=status_map.get(node.unique_id, NodeStatus.Skipped).value,
             )
             tests.append(test)
 
-            tests_dict[test.model].append(test)
+            if test.model:
+                tests_dict[test.model].append(test)
+            if test.source:
+                tests_dict[test.source].append(test)
```
Comment (chamini2) [inline src/faldbt/project.py:280]: curious on opinions here

Comment (isidentical) [reply src/faldbt/project.py:280]: I think it looks good!

**Pull request merged**